pkg/internal/controllers/test/review.go

```diff
@@ -0,0 +1,586 @@
+/*
```

Comment:
The tests seem really thorough to me 👍🏼 
Minor suggestion: perhaps some test cases could also include constraints?